### PR TITLE
code clean

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
move some comments to the appropriate place and remove some redundant.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -1705,33 +1704,29 @@ func buildCompleteFilterChain(pluginParams *plugin.InputParams, mutable *plugin.
...
+				mutable.Listener.FilterChains[i].Filters = append(mutable.Listener.FilterChains[i].Filters, chain.TCP...)
 			}
 			log.Debugf("attached %d network filters to listener %q filter chain %d", len(chain.TCP)+len(opt.networkFilters), mutable.Listener.Name, i)
 		} else {
 			// Add the TCP filters first.. and then the HTTP connection manager
-			if len(chain.TCP) > 0 {
-				mutable.Listener.FilterChains[i].Filters = append(mutable.Listener.FilterChains[i].Filters, chain.TCP...)
+			mutable.Listener.FilterChains[i].Filters = append(mutable.Listener.FilterChains[i].Filters, chain.TCP...)
+
+			opt.httpOpts.statPrefix = mutable.Listener.Name
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: I don't think this is safe, we can have nil httpOpts? https://github.com/istio/istio/blob/a05f491d85fb657fd1133e5a055c58a10e4f8aca/pilot/pkg/networking/core/v1alpha3/gateway.go#L533-L537

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: L1698 has already guaranteed it is not nil here.

**Pull request merged**